src/System.Diagnostics.Debug/src/System/Diagnostics/Debug.Windows.cs

```diff
@@ -0,0 +1,132 @@
+                        if (!System.Diagnostics.Debugger.IsAttached)
+                        {
+                            System.Diagnostics.Debugger.Launch();
+                        }
+                        System.Diagnostics.Debugger.Break();
+                        break;
+                    default:
+                        Debug.Assert(rval == Interop.User32.IDOK);
+                        break;
+                }
+            }
+
+            public void WriteLineCore(string message)
+            {
+                message = message + "\r\n";
+                Write(message);
+            }
+
+            public void WriteCore(string message)
+            {
+                // really huge messages mess up both VS and dbmon, so we chop it up into 
+                // reasonable chunks if it's too big. This is the number of characters 
+                // that OutputDebugstring chunks at.
+                const int WriteChunkLength = 4091;
+
+                // We don't want output from multiple threads to be interleaved.
+                lock (s_ForLock)
+                {
+                    if (message == null || message.Length <= WriteChunkLength)
+                    {
+                        WriteToDebugger(message);
+                    }
+                    else
+                    {
+                        int offset;
+                        for (offset = 0; offset < message.Length - WriteChunkLength; offset += WriteChunkLength)
+                        {
+                            WriteToDebugger(message.Substring(offset, WriteChunkLength));
+                        }
+                        WriteToDebugger(message.Substring(offset));
+                    }
+                }
+            }
+
+            [System.Security.SecuritySafeCritical]
+            private static void WriteToDebugger(string message)
+            {
+                Interop.mincore.OutputDebugString(message ?? string.Empty);
+            }
+
+            private static bool IsRTLResources
+            {
+                get
+                {
+                    return SR.RTL != "RTL_False";
+                }
+            }
+
+            internal class MessageBoxPopup
+            {
+                public int ReturnValue { get; set; }
+                private AutoResetEvent m_Event;
+                private string m_Body;
+                private string m_Title;
+                private int m_Flags;
+
+                [SecurityCritical]
+                public MessageBoxPopup(string body, string title, int flags)
+                {
+                    m_Event = new AutoResetEvent(false);
+                    m_Body = body;
+                    m_Title = title;
+                    m_Flags = flags;
+                }
+
+                public int ShowMessageBox()
+                {
+                    Thread t = new Thread(DoPopup);
+                    t.Start();
+                    m_Event.WaitOne();
```

Comment:
I'd guess this could also just be:
return Task.Run(() => MessageBox(...));
but I don't know if there's some reason it must be on a fresh Thread.